docs/source/algo_walkthrough.rst

```diff
@@ -0,0 +1,360 @@
+Odds are updated according to "*hits*" (where the range data is measured) and "*misses*" (the free space between the sensor and the measured points).
+Both *hits* and *misses* can have a different weight in occupancy probability calculations giving more or less trust to occupied or free space measurements.
+
+.. code-block:: lua
+
+    TRAJECTORY_BUILDER_2D.submaps.range_data_inserter.probability_grid_range_data_inserter.hit_probability
+    TRAJECTORY_BUILDER_2D.submaps.range_data_inserter.probability_grid_range_data_inserter.miss_probability
+    TRAJECTORY_BUILDER_3D.submaps.range_data_inserter.hit_probability
+    TRAJECTORY_BUILDER_3D.submaps.range_data_inserter.miss_probability
+
+In 2D, only one probability grid per submap is stored.
+In 3D, for scan matching performance reasons, two *hybrid* probability grids are used.
+(the term "hybrid" only refers to an internal tree-like data representation and is transparent to the user)
+
+- a low resolution hybrid grid for far measurements
+- a high resolution hybrid grid for close measurements
+
+Scan matching starts by aligning far points of the low resolution point cloud with the low resolution hybrid grid and then refines the pose by aligning the close high resolution points with the high resolution hybrid grid.
+
+.. code-block:: lua
+
+    TRAJECTORY_BUILDER_2D.submaps.grid_options_2d.resolution
+    TRAJECTORY_BUILDER_3D.submaps.high_resolution
+    TRAJECTORY_BUILDER_3D.submaps.low_resolution
+    TRAJECTORY_BUILDER_3D.high_resolution_adaptive_voxel_filter.max_range
+    TRAJECTORY_BUILDER_3D.low_resolution_adaptive_voxel_filter.max_range
+
+.. note::
+
+    Cartographer ROS provides an RViz plugin to visualize submaps. You can select the submaps you want to see from their number. In 3D, RViz only shows 2D projections of the 3D hybrid probability grids (in grayscale). Options are made available in RViz's left pane to switch between the low and high resolution hybrid grids visualization.
+
+**TODO**: *Documenting TSDF configuration*
+
+Global SLAM
+-----------
+
+While the local SLAM generates its succession of submaps, a global optimization (usually refered to as "*the optimization problem*" or "*sparse pose adjustment*") task runs in background.
+Its role is to re-arrange submaps between each other so that they form a coherent global map.
+For instance, this optimization is in charge of altering the currently built trajectory to properly align submaps with regards to loop closures.
+
+The optimization step should be done often enough to correct the global map in real time but there is no point running it more often than submaps construction because it works with submaps alignment.
+
+.. code-block:: lua
+
+    POSE_GRAPH.optimize_every_n_nodes
+
+.. note::
+
+    Setting POSE_GRAPH.optimize_every_n_nodes to 0 is a handy way to disable Global SLAM and concentrate on the behavior of Local SLAM. This is usually one of the first thing to do to tune Cartographer.
```

Comment:
`Local SLAM` or `local SLAM`? just check if it's consistent in the document (also global SLAM and other terms)